packages/build-tools/src/ios/pod.ts

```diff
@@ -1,16 +1,50 @@
 import path from 'path';
 
+import fs from 'fs-extra';
 import { Ios } from '@expo/eas-build-job';
 import spawn from '@expo/turtle-spawn';
 
 import { BuildContext } from '../context';
 
 export async function installPods<TJob extends Ios.Job>(ctx: BuildContext<TJob>): Promise<void> {
   const iosDir = path.join(ctx.reactNativeProjectDirectory, 'ios');
-  ctx.logger.info('Installing pods');
+
+  if (ctx.env.EAS_BUILD_COCOAPODS_CACHE_URL) {
+    ctx.logger.info('Running using experimental CocoaPods cache proxy...');
+    const podfilePath = path.join(iosDir, 'Podfile');
+    const originalPodfileContents = await fs.readFile(podfilePath, 'utf-8');
+
+    const cocoaPodsCdnSourceRegex = /source( )+('|")https:\/\/cdn.cocoapods.org(\/)*('|")/;
```

Comment:
I think that would be more accurate and handles the invocation with parentheses (in Ruby we usually skip them though).

```suggestion
    const cocoaPodsCdnSourceRegex = /\bsource\s*\(?['"]https:\/\/cdn\.cocoapods\.org\/?['"]\)?/g;
```